examples/sending_concurrent_transactions/main.go

```diff
@@ -0,0 +1,143 @@
+func setup(networkConfig aptos.NetworkConfig) (*aptos.Client, aptos.TransactionSigner) {
+	client, err := aptos.NewClient(networkConfig)
+	if err != nil {
+		panic("Failed to create client:" + err.Error())
+	}
+
+	sender, err := aptos.NewEd25519Account()
+	if err != nil {
+		panic("Failed to create sender:" + err.Error())
+	}
+
+	err = client.Fund(sender.Address, 100_000_000)
+	if err != nil {
+		panic("Failed to fund sender:" + err.Error())
+	}
+
+	return client, sender
+}
+
+func args() [][]byte {
+	receiver := aptos.AccountAddress{}
+	err := receiver.ParseStringRelaxed("0xBEEF")
+	if err != nil {
+		panic("Failed to parse address:" + err.Error())
+	}
+	amount := uint64(100)
+
+	receiverArg, err := bcs.Serialize(&receiver)
+	amountArg, err := bcs.SerializeU64(amount)
+	if err != nil {
+		panic("Failed to serialize arguments:" + err.Error())
+	}
+	return [][]byte{receiverArg, amountArg}
+}
+
+func payload() aptos.TransactionPayload {
+	return aptos.TransactionPayload{Payload: &aptos.EntryFunction{
+		Module: aptos.ModuleId{
+			Address: aptos.AccountOne,
+			Name:    "aptos_account",
+		},
+		Function: "transfer",
+		ArgTypes: []aptos.TypeTag{},
+		Args:     args(),
+	}}
+}
+
+func sendManyTransactionsSerially(networkConfig aptos.NetworkConfig, numTransactions uint64) {
+	client, sender := setup(networkConfig)
+
+	responses := make([]*api.SubmitTransactionResponse, numTransactions)
+	payload := payload()
+
+	senderAddress := sender.AccountAddress()
+	sequenceNumber := uint64(0)
+	for i := uint64(0); i < numTransactions; i++ {
+		rawTxn, err := client.BuildTransaction(senderAddress, payload, aptos.SequenceNumber(sequenceNumber))
+		if err != nil {
+			panic("Failed to build transaction:" + err.Error())
+		}
+
+		signedTxn, err := rawTxn.SignedTransaction(sender)
+		if err != nil {
+			panic("Failed to sign transaction:" + err.Error())
+		}
+
+		submitResult, err := client.SubmitTransaction(signedTxn)
+		if err != nil {
+			panic("Failed to submit transaction:" + err.Error())
+		}
+		responses[i] = submitResult
+		sequenceNumber++
+	}
+
+	// Wait on last transaction
+	response, err := client.WaitForTransaction(responses[numTransactions-1].Hash)
+	if err != nil {
+		panic("Failed to wait for transaction:" + err.Error())
+	}
+	if response.Success == false {
+		panic("Transaction failed due to " + response.VmStatus)
+	}
+}
+
+func sendManyTransactionsConcurrently(networkConfig aptos.NetworkConfig, numTransactions uint64) {
+	client, sender := setup(networkConfig)
+	payload := payload()
+
+	// start submission goroutine
+	payloads := make(chan aptos.TransactionBuildPayload, 50)
+	results := make(chan aptos.TransactionSubmissionResponse, 50)
+	go client.BuildSignAndSubmitTransactions(sender, payloads, results)
+
+	// Submit transactions to goroutine
+	for i := uint64(0); i < numTransactions; i++ {
+		payloads <- aptos.TransactionBuildPayload{
+			Id:    i,
+			Type:  aptos.TransactionSubmissionTypeSingle,
+			Inner: payload,
+		}
+	}
+	close(payloads)
```

Comment:
this should happen in its own `go func() { ... }()` block, if numTransactions is very much more than 100 input and submission could get blocked waiting for results to be consumed below